lib/src/kbasesearchengine/tools/SearchTools.java

```diff
@@ -223,10 +223,27 @@ public int execute() {
     }
     
     private void waitForReturn(final Stoppable stoppable) throws InterruptedException {
-        out.println("Press return to shut down process.");
-        console.readLine();
-        System.out.println("Waiting for task to stop");
-        stoppable.stop(10 * 60 * 1000); // 10 mins
+        if (console == null) {
```

Comment:
Don't you also need to deal with the nonNull() call on line 111 so that it doesn't abort at startup?

Reply:
yep, forgot. fixed